### PR TITLE
feat: Add calendar alert

### LINKED ISSUE #1164: Add to Calendar in Proposal view
How about we get rid of the "add to watchlist" feature and push this feature instead?

Requirements:
- modal displayed after voting to suggest creating calendar alert instead of adding to watchlist.
- prompt user to select how long before they'd want the alert to happen; allow for minutes/hours/days; allow for 1 decimal (ie. 1.5 hours).
- by clicking on the "add to calendar" CTA the user is brought to Google Calendar to actually finish submitting
- there will be no success instance on the governance platform when successfully adding an alert, and this should be acceptable

Image

Image

### PR DESCRIPTION
**Pull request by ncomerci**
This closes #1164 

image
image

### CODE REVIEW

File: src/components/Proposal/View/CalendarAlertButton.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import useFormatMessage from '../../../hooks/useFormatMessage'
+import CalendarAdd from '../../Icon/CalendarAdd'
+
+interface Props {
+  loading?: boolean
+  disabled?: boolean
+  onClick: () => void
+}
+
+function CalendarAlertButton({ loading, disabled, onClick }: Props) {
```
Comment (andyesp) [inline src/components/Proposal/View/CalendarAlertButton.tsx:15]: Can't we reuse the `SidebarLinkButton` component?

File: src/components/ProjectRequest/NumberSelector.tsx [hunk 2]
```diff
@@ -34,10 +40,30 @@ const NumberSelector = ({ value, onChange, min, max, label, unitLabel, subtitle
...
+
+  const getUnitLabel = (unit: UnitTypeLongPlural) => t(`general.time_units.${unit}`)
+  const getUnitOptions = (units: UnitTypeLongPlural[]) =>
+    units.map((unit) => ({ key: unit, value: unit, text: getUnitLabel(unit) }))
+
   return (
     <div className="NumberSelector">
       <Label>{label}</Label>
-      <div>
+      <div className="NumberSelectorContainer">
```
Comment (andyesp) [inline src/components/ProjectRequest/NumberSelector.tsx:66]: `NumberSelector__Container`

File: src/components/Modal/CalendarAlertModal.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+function CalendarAlertModal({ proposal, onClose, ...props }: CalendarAlertModalProps) {
+  const t = useFormatMessage()
+  const { finish_at, title, id } = proposal
+  const [value, setValue] = useState(0)
+  const [unit, setUnit] = useState<UnitTypeLongPlural | undefined>()
+  const [isDisabled, setIsDisabled] = useState(false)
+
+  const handleOnClose = () => {
```
Comment (andyesp) [inline src/components/Modal/CalendarAlertModal.tsx:41]: `handleClose`

File: src/components/Modal/CalendarAlertModal.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  if (alertDate.isBefore(currentDate)) {
+    return null
+  }
+  return alertDate.toDate()
+}
+
+function CalendarAlertModal({ proposal, onClose, ...props }: CalendarAlertModalProps) {
+  const t = useFormatMessage()
+  const { finish_at, title, id } = proposal
+  const [value, setValue] = useState(0)
```
Comment (1emu) [inline src/components/Modal/CalendarAlertModal.tsx:37]: what is value? 
baby don't hurt me

File: src/components/Modal/CalendarAlertModal.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    <Modal {...props} onClose={handleOnClose} size="tiny" closeIcon={<Close />}>
+      <Modal.Content>
+        <div className="ProposalModal__Title">
+          <Header>{t('modal.calendar_alert.title')}</Header>
+        </div>
+        <div className="CalendarAlertModal__SelectorContainer">
+          <NumberSelector
+            value={value}
+            min={0}
+            max={300}
```
Comment (1emu) [inline src/components/Modal/CalendarAlertModal.tsx:72]: what is 300? 

image

**Review by 1emu — APPROVED** [review]

**Review by andyesp — APPROVED** [review]

**Pull request merged**